packages/bdd-tests/services/getaccessToken.ts

```diff
@@ -1,64 +1,257 @@
-import axios from "axios"
-import qs from 'qs'
-import {Req}  from '../src/configs/spec'
-import DomParser from 'dom-parser'
+/**
+ * TypeScript version of `identity_service.py` from pytest-nhsd-apim
+ * https://github.com/NHSDigital/pytest-nhsd-apim/blob/82316e098a8696e4f7b0de7747b580dc0ffa7229/
+ * /src/pytest_nhsd_apim/identity_service.py#L298
+ *
+ * 1. Hit `authorize` endpoint w/ required query params --> we
+ * are redirected to the simulated_auth page. The requests package
+ * follows those redirects.
+ *
+ * 2. Parse the login page.  For keycloak this presents an
+ * HTML form, which must be filled in with valid data.  The tester
+ * can submits their login data with the `login_form` field.
+ *
+ * 3. POST the filled in form. This is equivalent to clicking the
+ * "Login" button if we were a human.
+ *
+ * 4. The mock auth redirected us back to the
+ * identity-service, which redirected us to whatever our app's
+ * callback-url was set to.  We don't actually care about the
+ * content our callback-url page, we just need the auth_code that
+ * was provided in the redirect.
+ *
+ * 5. Finally, get an access token.
+ *
+ * 6. Profit... sweet sweet profit.
+ */
+import axios, {AxiosInstance, AxiosResponse} from "axios"
+import {JSDOM} from "jsdom"
+import {wrapper} from "axios-cookiejar-support"
+import {CookieJar} from "tough-cookie"
+import {parse} from "query-string"
 
-let resp;
-let code;
+export const VALID_APIGEE_ENVIRONMENTS = ["internal-dev", "internal-qa", "int", "ref"]
 
+type EnvironmentSecrets = {
+  clientId: string
+  clientSecret: string
+}
+
+export class AuthClient {
+  private readonly environment: string
+  readonly clientId: string
+  readonly clientSecret: string
+  readonly callbackUrl: string
+  private readonly state: string
+
+  constructor() {
+    const {clientId, clientSecret} = this.getApiCredentials()
+    this.clientId = clientId
+    this.clientSecret = clientSecret
+
+    this.environment = this.getEnvironment()
+    this.callbackUrl = "https://example.org/" // using /callback causes the website to return a 404, which upsets axios
+    this.state = this.getState()
+  }
+
+  getBaseUrl(): string {
+    return `https://${this.environment}.api.service.nhs.uk/oauth2-mock`
+  }
 
-const config1 = {
-  params: {
-    client_id: process.env.client_id,
-    redirect_uri: process.env.redirect_uri,
-    state: process.env.state,
-    response_type: process.env.response_type,
+  getKeycloakUrl(): string {
+    return `https://identity.ptl.api.platform.nhs.uk/auth/realms/Cis2-mock-internal-dev/protocol/openid-connect`
   }
-};
 
-const cookieJar = {
-  myCookies: undefined,
-};
+  private getState(): string {
+    // return uuid.v4()
```

Comment:
Does this want un-commenting and the below line removing?

Also, this file should probably be called getAccessToken.ts

Reply:
renamed file
The file is just a copy of packages/e2e-tests/auth/src/oauth.ts so that included the commented out line as well